rust/lang/security/audit/command-injection-process-builder.yaml

```diff
@@ -0,0 +1,26 @@
+rules:
+  - id: dangerous-spawn-process
+    message: |
+      Found dynamic content when spawning a process. This is dangerous if external
+      data can reach this function call because it allows a malicious actor to
+      execute commands. Ensure no external data reaches here.
+    metadata:
+      cwe: "CWE-78: Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection')"
+      owasp: 'A1: Injection'
+      asvs:
+        section: 'V5: Validation, Sanitization and Encoding Verification Requirements'
+        control_id: 5.3.8 OS Command Injection
+        control_url: https://github.com/OWASP/ASVS/blob/master/4.0/en/0x13-V5-Validation-Sanitization-Encoding.md#v53-output-encoding-and-injection-prevention-requirements
+        version: '4'
+    languages: [rust]
+    severity: WARNING
+    patterns:
+      - pattern-either:
+          - patterns:
+              - pattern: |
+                  fn $FUNC(...) {
+                    ...
+                    Command::new(...);
+                    ...
+                  }
+
```

Comment:
```suggestion
              - pattern: |
                    Command::new(...)
              - pattern-not: |
                    Command::new("...")

```

Some suggestions:
1. By removing the function declaration around the actual vulnerable line of code we can reduce the amount of lines being reported or marked. Instead of the entire function declaration, Semgrep will now report the exact line of the `Command::new`. You will need to update the test syntax accordingly! Additionally, this would also capture calls that do not end in a semicolon. 

For example, a construction like this might be missed by your pattern (I didn't actually run it):
```
let output = if cfg!(target_os = "windows") {
    Command::new(cmd)
            .args(["/C", "echo hello"])
            .output()
            .expect("failed to execute process")
} else {
    Command::new("sh")
            .arg("-c")
            .arg("echo hello")
            .output()
            .expect("failed to execute process")
};
```

You should look up some uses of the functions and add them to the testcode with appropriate test syntax to verify this!

2. I have added a pattern-not to exclude very obvious False Positives, like a hardcoded String. This will reduce the noise for this rule.